packages/keystone/apolloErrorFormatter.spec.js

```diff
@@ -287,13 +309,740 @@ describe('safeFormatError hide=false', () => {
+        const error = new GraphQLError(message2, null, null, null, null, original, {})
+        const result = safeFormatError(error)
+        const uid = result?.originalError?.uid
+        expect(result).toEqual({
+            'name': 'GQLError',
+            'message': message2,
+            'fullstack': expect.stringMatching(new RegExp(`^GQLError: ${message1}(.*?)Caused By: GQLError: ${message3}(.*?)Caused By: Error: World`, 's')),
+            'stack': expect.stringMatching(new RegExp(`^GQLError: ${message1}`)),
+            'extensions': { ...fields1, 'message': message3 },
+            'originalError': {
+                'name': 'GQLError',
+                'message': message1,
+                'stack': expect.stringMatching(new RegExp(`^GQLError: ${message1}`)),
+                'extensions': { ...fields1, 'message': message1 },
+                'errors': [
+                    {
+                        'name': 'GQLError',
+                        'message': message3,
+                        'stack': expect.stringMatching(new RegExp(`^GQLError: ${message3}`)),
+                        'extensions': { ...fields2, 'message': message3 },
+                        'errors': [{
+                            'message': 'World',
+                            'name': 'Error',
+                            'stack': expect.stringMatching(new RegExp('^Error: World')),
+                        }],
+                    },
+                ],
+            },
+        })
+    })
+    test('safeFormatError(GQLError) with nested errors', () => {
+        const message1 = Date.now().toString()
+        const message2 = 'GQL' + (Date.now() % 100).toString()
+        const message3 = 'ERR' + (Date.now() % 800).toString()
+        const message4 = 'XXX' + (Date.now() % 900).toString()
+        const fields2 = {
+            code: 'BAD_USER_INPUT',
+            type: 'WRONG_FORMAT',
+        }
+        const fields1 = {
+            code: 'INTERNAL_ERROR',
+            type: 'SUB_GQL_ERROR',
+        }
+        const original2 = new Error('InError')
+        original2.errors = [new GQLError({ ...fields1, message: message4 }, null)]
+        const validation = new Error('KeystoneValidation')
+        validation.errors = [new GraphQLError(message1, null, null, null, null, new GQLError({ ...fields2, message: message3 }, null, original2), {})]
+        const error = new GraphQLError(message2, null, null, null, null, validation, {})
+        const result = safeFormatError(error)
+        expect(result).toEqual({
+            'name': 'GraphQLError',
+            'message': message2,
+            'stack': expect.stringMatching(new RegExp('^Error: KeystoneValidation')),
+            'fullstack': expect.stringMatching(new RegExp(`^Error: KeystoneValidation(.*?)Caused By: GQLError: ${message3}(.*?)Caused By: Error: InError(.*?)Caused By: GQLError: ${message4}`, 's')),
+            'originalError': {
+                'message': 'KeystoneValidation',
```

Comment:
Seems like this part should go to name, and message is something with more details?

Reply:
In this case no. It's non real case. I've a ValidationFailureError for that